openerp/addons/base/ir/ir_model.py

```diff
@@ -237,6 +237,45 @@ class CustomModel(models.Model):
 
         CustomModel._build_model(self.pool, cr)
 
+    def load(self, cr, uid, fields, data, context=None):
+        """
+        Attempts to load the data matrix, and returns a list of ids (or
+        ``False`` if there was an error and no id could be generated) and a
+        list of messages.
+
+        The ids are those of the records created and saved (in database), in
+        the same order they were extracted from the file. They can be passed
+        directly to :meth:`~read`
+
+        :param fields: list of fields to import, at the same index as the corresponding data
+        :type fields: list(str)
+        :param data: row-major matrix of data to import
+        :type data: list(list(str))
+        :param dict context:
+        :returns: {ids: list(int)|False, messages: [Message]}
+        """
+        context = dict(context or {}, manual=True)
+        fields = map(models.fix_import_export_id_paths, fields)
+        ModelData = self.pool['ir.model.data']
+
+        ids = []
+        messages = []
+        for id, xid, record, info in self._convert_records(cr, uid,
+                self._extract_records(cr, uid, fields, data, context=context, log=messages.append),
+                context=context, log=messages.append):
+
+            if not id:
+                data_ids = ModelData.search_read(cr, uid, [('name', '=', xid)], ['res_id'], context=context)
+                id = data_ids and data_ids[0] and data_ids[0]['res_id'] or None
```

Comment:
I suggest you use `xmlid_to_res_id` instead:

    id = ModelData.xmlid_to_res_id(cr, uid, xid)